std/conv.d

```diff
@@ -2341,13 +2341,15 @@ in
 body
 {
     import core.checkedint : mulu, addu;
+    import std.exception : enforce;
+
+    enforce!ConvException(!s.empty, "s must not be empty in integral parse");
```

Comment:
I moved the logic because `atStart` was just a poor-man's `range.empty`. It makes much more sense to do sanity checks on input at the start of the function.